ecommerce/backend/API/tests/test_recommendation.py

```diff
@@ -0,0 +1,76 @@
+        p1 = Product.objects.create(id=2, name = "Mavi T-shirt3", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s, brand = b, vendor = v)
+        p2 = Product.objects.create(id=3, name = "Mavi T-shirt4", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s, brand = b, vendor = v)
+        p3 = Product.objects.create(id=4, name = "Mavi T-shirt5", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s, brand = b, vendor = v)
+        p4 = Product.objects.create(id=5, name = "Mavi T-shirt6", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s, brand = b, vendor = v)
+        Address.objects.create(id=2,user=user, title=f"Address-1", name="OmerFaruk", surname="Deniz", 
+            address= "Mahalle Sokak Sk. No 23/8", province= "Sarıyer", city= "Istanbul", 
+            phone_country_code= "+90", phone_number= "5351234567", country= "Turkey", zip_code= "34344")
+        p21 = Product.objects.create(id=6, name = "other1", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s2, brand = b, vendor = v)
+        p22 = Product.objects.create(id=7, name = "other2", price = 100, 
+            creation_date = "2019-08-20T07:22:34Z",total_rating = 20, discount=0.1,
+            rating_count = 5, stock_amount = 10, short_description = "yaza özel",long_description = "gerçekten yaza özel yav", subcategory = s2, brand = b, vendor = v)
+        order = Order.objects.create(id=1, user_id=1)
+        purchase = Purchase.objects.create(id=1,amount=3,unit_price=22.11,status=0,purchase_date='2019-08-20 10:22:34+03',address_id=2,product=p,vendor=v,order_id=1)
+        order = Order.objects.create(id=2, user_id=1)
+        purchase = Purchase.objects.create(id=2,amount=3,unit_price=22.11,status=0,purchase_date='2019-08-20 10:22:34+03',address_id=2,product=p21,vendor=v,order_id=2)
+
+    @tag('recommend')
+    def test_1_recommendation(self):
+        response = self.client.get(reverse(recommend_products))
+
+        self.assertEqual(len(response.data["products"]), 2)
```

Comment:
please also chech for 200 and status in the response:
```suggestion
        self.assertEqual(len(response.data["products"]), 2)
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.data["status"]["successful"], True)
```

Reply:
Done